### PR TITLE
get module name from sys.modules

### PR DESCRIPTION
**Pull request by girarda**
## What
This is a fix for https://github.com/airbytehq/airbyte/issues/17772.
Acceptance tests don't run from the same main file as running the discovery command.
Instead of inferring the connector's module from the main file, we can get it from the list of modules that are loaded in the environment at runtime

## How
* Iterate over the modules loaded at runtime and select the one that starts with "source_"

## Recommended reading order
1. `airbyte-cdk/python/airbyte_cdk/sources/declarative/schema/json_schema.py`

## Testing
Testing cdk changes is tedious. I used the following script with the following docker image to run the SATs using this branch

```
CURRENT_DIR="$PWD"
DIR=$(basename "$PWD")
echo $DIR

# Build SAT
cd ../../bases/source-acceptance-test
docker build . -t airbyte/source-acceptance-test:dev

# Build connector
cd ../../../
docker build . -t airbyte/$DIR:dev -f airbyte-integrations/connectors/$DIR/Dockerfile
cd $CURRENT_DIR
#python3 -m pytest integration_tests -p integration_tests.acceptance

# Run
docker run --rm -it \
    -v /var/run/docker.sock:/var/run/docker.sock \
    -v /tmp:/tmp \
    -v $(pwd):/test_input \
    --env CASSETTE_MODE \
    --env CASSETTE_PATH \
    airbyte/source-acceptance-test:dev \
    --acceptance-test-config /test_input
```

dockerfile:
```
FROM python:3.9-slim

# Bash is installed for more convenient debugging.
RUN apt-get update && apt-get install -y bash automake build-essential git libtool libtool-bin autoconf && rm -rf /var/lib/apt/lists/*

WORKDIR /airbyte/integration_code
COPY ./airbyte-integrations/connectors/source-exchange-rates-tutorial-2/source_exchange_rates_tutorial_2 ./source_exchange_rates_tutorial_2
RUN echo "$ls"

COPY ./airbyte-cdk/python ./cdk

COPY ./airbyte-integrations/connectors/source-exchange-rates-tutorial-2/main.py ./
COPY ./airbyte-integrations/connectors/source-exchange-rates-tutorial-2/setup.py ./

#COPY setup.py ./
RUN pip install -e ./cdk
RUN pip install .

ENV AIRBYTE_ENTRYPOINT "python /airbyte/integration_code/main.py"
ENTRYPOINT ["python", "/airbyte/integration_code/main.py"]

LABEL io.airbyte.version=0.3.1
LABEL io.airbyte.name=airbyte/source-exchange-rates-tutorial-2
```

## 🚨 User Impact 🚨
Are there any breaking changes? What is the end result perceived by the user? If yes, please merge this PR with the 🚨🚨 emoji so changelog authors can further highlight this if needed.

## Pre-merge Checklist
Expand the relevant checklist and delete the others.

<details><summary><strong>New Connector</strong></summary>

### Community member or Airbyter

- [ ] **Community member?** Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret`
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
    - [ ] `docs/integrations/README.md`
    - [ ] `airbyte-integrations/builds.md`
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.com/contributing-to-airbyte/issues-and-pull-requests)

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)
- [ ] After the connector is published, connector added to connector index as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
- [ ] Seed specs have been re-generated by building the platform and committing the changes to the seed spec files, as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)

</details>

<details><summary><strong>Updating a connector</strong></summary>

### Community member or Airbyter

- [ ] Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret`
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] Changelog updated in `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.com/contributing-to-airbyte/issues-and-pull-requests)

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing
- [ ] New Connector version released on Dockerhub and connector version bumped by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)

</details>

<details><summary><strong>Connector Generator</strong></summary>

- [ ] Issue acceptance criteria met
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.com/contributing-to-airbyte/issues-and-pull-requests)
- [ ] If adding a new generator, add it to the [list of scaffold modules being tested](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connector-templates/generator/build.gradle#L41)
- [ ] The generator test modules (all connectors with `-scaffold` in their name) have been updated with the latest scaffold by running `./gradlew :airbyte-integrations:connector-templates:generator:testScaffoldTemplates` then checking in your changes
- [ ] Documentation which references the generator is updated as needed

</details>

## Tests

<details><summary><strong>Unit</strong></summary>

*Put your unit tests output here.*

</details>

<details><summary><strong>Integration</strong></summary>

*Put your integration tests output here.*

</details>

<details><summary><strong>Acceptance</strong></summary>

*Put your acceptance tests output here.*

</details>

### CODE REVIEW

File: airbyte-cdk/python/airbyte_cdk/sources/declarative/schema/json_schema.py [hunk 1]
```diff
@@ -4,19 +4,23 @@
...
 
 
 def _default_file_path() -> str:
-    main_file = __main__.__file__
-    module = main_file.split("/")[-2].replace("-", "_")
+    # schema files are always in "source_<connector_name>/schemas/<stream_name>.json
+    # the connector's module name can be inferred by looking at the modules loaded and look for the one starting with source_
+    source_modules = [
+        k for k, v in sys.modules.items() if "source_" in k
+    ]  # example: ['source_exchange_rates', 'source_exchange_rates_tutorial.source']
```
Comment (brianjlai) [inline airbyte-cdk/python/airbyte_cdk/sources/declarative/schema/json_schema.py:22]: I'm a little confused by your comment example here. Are you saying that the result of one `sys.modules` call will result in these two items in the map? Or that these are two possible permutations of how the `sys.modules` call will be returned (one called from the `source_exchange_rates` source and one called from the `source_exchange_rates_tutorial` source)?

**Review by brianjlai — APPROVED** [review]
a few small things, but don't need to block on approval

File: airbyte-cdk/python/airbyte_cdk/sources/declarative/schema/json_schema.py [hunk 2]
```diff
@@ -4,19 +4,23 @@
...
+    module = source_modules[0].split(".")[0]
```
[the remark this answers is not in the document]
Comment (girarda) [reply airbyte-cdk/python/airbyte_cdk/sources/declarative/schema/json_schema.py:25]: done

File: airbyte-cdk/python/airbyte_cdk/sources/declarative/schema/json_schema.py (hunk 1 above)
Comment (girarda) [reply airbyte-cdk/python/airbyte_cdk/sources/declarative/schema/json_schema.py]: fixed the comment. thanks!

**Pull request merged**